src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDirective.java

```diff
@@ -85,16 +85,22 @@ void putMetric(String key, double value, Unit unit, StorageResolution storageRes
         metrics.compute(
                 key,
                 (k, v) -> {
-                    if (v == null) return new MetricDefinition(key, unit, storageResolution, value);
-                    else {
-                        v.addValue(value);
+                    if (v == null) {
+                        MetricDefinitionBuilder tmp =
+                                new MetricDefinitionBuilder(unit, storageResolution, value);
+                        tmp.setName(key);
```

Comment:
Why don't we allow name to be part of constructor?

Reply:
Refactored to no longer be an issue